application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/reportcalculation/controller/ReportCalculationController.kt

```diff
@@ -26,17 +33,30 @@ class ReportCalculationController(
 ) {
     @PostMapping("/report/{reportId}")
     fun startCalculation(
-        @PathVariable reportId: String
+        @PathVariable reportId: String,
+        @RequestParam @DateTimeFormat(pattern = "yyyy-MM-dd") from: Date?,
+        @RequestParam @DateTimeFormat(pattern = "yyyy-MM-dd") to: Date?,
     ): Mono<DomainReference> {
         return reportingStorage.fetchReport(DomainReference(id = reportId))
             .flatMap { reportOptional ->
                 val report = reportOptional.orElseThrow {
                     NotFoundException()
                 }
 
-                createReportCalculationUseCase.startReportCalculation(
+                val args = mutableMapOf<String, String>()
+
+                if (from != null) {
+                    args["from"] = from.toInstant().atOffset(ZoneOffset.UTC).format(DateTimeFormatter.ISO_DATE_TIME)
```

Comment:
what does the offset do here? We have had issues earlier with dates shifting a few hours into another day due to timezone. The expected behavior here should be that submitting a param YYYY-MM-dd from anywhere in the world will result in the same report result

Reply:
I would like to have all data in the database always in the UTC timezone. No matter what timezone the server is in. On Controller level, we just accept a date in this format: "yyyy-MM-dd", and will be transformed into an ISO_DATE.

from: 2022-02-02 will be 2022-02-02T00:00:000Z in the database
to: 2022-02-02 will be 2022-02-02T23:59:999Z in the database

So it does not matter wich timezone the user is in.

Reply:
totally agree regarding the universal db persisted format.

My concern is the following use cases (as we have had a lot of chaos regarding time zones in reports previously). Maybe you can confirm these work correctly with this:

scenario:
- event recorded in the US (GMT-7) at 21:00 local time (in CouchDB we intentionally store only the date of events, without timezone)
- user A in US running report
- user B in Germany running report

1. Through our app, as a user in Germany, I select start and end date (e.g. until last Saturday) and get a report that includes all events on these days according to local time of my team in the US (e.g. including an event that was recorded at 21:00 US time = 04:00 GMT = 06:00 German time)
2. Through our API, the same is true and I do not have to worry about converting the parameters of the query to the "right" timezone ... (:question: what does "not have to worry" mean? From experience, I would ideally send only a date without timezone as params, because that explicitly removes that questions here)